src/runtime/virtcontainers/proxy.go

```diff
@@ -1,270 +0,0 @@
-		if err != nil {
-			return "", err
-		}
-		socketPath := filepath.Join(filepath.Join(store.RunStoragePath(), id), "proxy.sock")
-		return fmt.Sprintf("unix://%s", socketPath), nil
-	case SocketTypeVSOCK:
-		// TODO Build the VSOCK default URL
-		return "", nil
-	default:
-		return "", fmt.Errorf("Unknown socket type: %s", socketType)
-	}
-}
-
-func isProxyBuiltIn(pType ProxyType) bool {
-	return pType == KataBuiltInProxyType
-}
-
-// proxy is the virtcontainers proxy interface.
-type proxy interface {
-	// start launches a proxy instance with specified parameters, returning
-	// the PID of the process and the URL used to connect to it.
-	start(params proxyParams) (int, string, error)
-
-	// stop terminates a proxy instance after all communications with the
-	// agent inside the VM have been properly stopped.
-	stop(pid int) error
-
-	//check if the proxy has watched the vm console.
-	consoleWatched() bool
-}
-
-func (p *proxyBuiltin) watchConsole(proto, console string, logger *logrus.Entry) (err error) {
-	var (
-		scanner *bufio.Scanner
-		conn    net.Conn
-	)
-
-	switch proto {
-	case consoleProtoUnix:
-		conn, err = net.Dial("unix", console)
-		if err != nil {
-			return err
-		}
-		// TODO: please see
-		// https://github.com/kata-containers/runtime/issues/1940.
-	case consoleProtoPty:
-		fallthrough
-	default:
-		return fmt.Errorf("unknown console proto %s", proto)
-	}
-
-	p.conn = conn
-
-	go func() {
-		scanner = bufio.NewScanner(conn)
-		for scanner.Scan() {
-			logger.WithFields(logrus.Fields{
-				"sandbox":   p.sandboxID,
-				"vmconsole": scanner.Text(),
-			}).Debug("reading guest console")
-		}
-
-		if err := scanner.Err(); err != nil {
-			if err == io.EOF {
-				logger.Info("console watcher quits")
-			} else {
-				logger.WithError(err).WithFields(logrus.Fields{
-					"console-protocol": proto,
-					"console-socket":   console,
-				}).Error("Failed to read agent logs")
-			}
-		}
-	}()
-
-	return nil
-}
-
-// check if the proxy has watched the vm console.
-func (p *proxyBuiltin) consoleWatched() bool {
-	return p.conn != nil
-}
-
-// start is the proxy start implementation for builtin proxy.
-// It starts the console watcher for the guest.
-// It returns agentURL to let agent connect directly.
-func (p *proxyBuiltin) start(params proxyParams) (int, string, error) {
-	if params.logger == nil {
-		return -1, "", fmt.Errorf("Invalid proxy parameter: proxy logger is not set")
-	}
-
-	if p.consoleWatched() {
-		return -1, "", fmt.Errorf("The console has been watched for sandbox %s", params.id)
-	}
-
-	params.logger.Debug("Start to watch the console")
-
-	p.sandboxID = params.id
-
-	// For firecracker, it hasn't support the console watching and it's consoleURL
-	// will be set empty.
-	// TODO: add support for hybrid vsocks, see https://github.com/kata-containers/runtime/issues/2098
-	if params.debug && params.consoleURL != "" && !strings.HasPrefix(params.consoleURL, kataclient.HybridVSockScheme) {
```

Comment:
We should keep watching console and use something like `params.debug` to control it.

Reply:
Yeah makes sense.. console is really useful for debugging.